### PR TITLE
8.2 XS82ECU1063

### PR DESCRIPTION
**Pull request by gduperrey**
Koji build scratch to v8.2-incoming: https://koji.xcp-ng.org/taskinfo?taskID=54241

### CODE REVIEW

File: SPECS/curl.spec [hunk 1]
```diff
@@ -322,7 +309,7 @@ rm -f ${RPM_BUILD_ROOT}%{_libdir}/libcurl.la
 # should be fixed in next release https://github.com/curl/curl/pull/12843
 rm -f ${RPM_BUILD_ROOT}%{_mandir}/man1/mk-ca-bundle.1*
 
-%ldconfig_scriptlets -n libcurl
+/sbin/ldconfig -n libcurl
```
Comment (stormi) [inline SPECS/curl.spec]: Where does this come from? It looks wrong to do this in %install.

File: SPECS/curl.spec [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+%{_libdir}/*.so
+%{_libdir}/pkgconfig/*.pc
+%{_mandir}/man1/curl-config.1*
+%{_mandir}/man3/*
+%{_datadir}/aclocal/libcurl.m4
+
+%changelog
+* Wed May 22 2024 Gael Duperrey <gduperrey@vates.tech> - 8.6.0-2.1
+- Synced from XS82ECU1063
+- Removed xenserver-specific test of %dist to determine buildrequires
```
Comment (stormi) [inline SPECS/curl.spec]: Incomplete. It's not just for the buildrequires. Better not describe what the test did, rather than give an incomplete description (here it also affects enabling or disabling an option of the build that is not directly related to buildrequires)

And you didn't escape %dist.

```suggestion
- Removed xenserver-specific test of the dist macro
```

File: SPECS/curl.spec [hunk 3]
```diff
@@ -353,6 +340,10 @@ rm -f ${RPM_BUILD_ROOT}%{_mandir}/man1/mk-ca-bundle.1*
 %{_datadir}/aclocal/libcurl.m4
 
 %changelog
+* Wed May 22 2024 Gael Duperrey <gduperrey@vates.tech> - 8.6.0-2.1
```
Comment (stormi) [inline SPECS/curl.spec]: I think you can update the date to May 29.

File: SPECS/curl.spec (hunk 1 above)
Comment (gduperrey) [reply SPECS/curl.spec]: Because it failed if we didn't do like this. @tescande helped me on this one.

Comment (stormi) [reply SPECS/curl.spec]: Well, it's wrong :)

Comment (gduperrey) [reply SPECS/curl.spec]: And do you have any suggestion on this change as the initial line doesn't work?

Comment (stormi) [reply SPECS/curl.spec]: Yes, but I would prefer not to be the one who explains why it's wrong to replace this macro without knowing what it used to do.

Comment (stormi) [reply SPECS/curl.spec]: And `/sbin/ldconfig -n libcurl` clearly makes no sense.

Comment (stormi) [reply SPECS/curl.spec]: Check what the older curl SRPM did with ldconfig...

Comment (tescande) [reply SPECS/curl.spec]: Ok, so my bad. I thought the install steps needed to refresh the so cache before continuing the installation.

So the solution is to replace the macro `%ldconfig_scriptlets` with:
```
%post -n libcurl -p /sbin/ldconfig
%postun -n libcurl -p /sbin/ldconfig
```

Edit: Looks like `-n` is also needed for `%postun`

Comment (gduperrey) [reply SPECS/curl.spec]: I allready changed it with the entry in the older srpm from koji.

File: SPECS/curl.spec [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+%{_libdir}/*.so
+%{_libdir}/pkgconfig/*.pc
+%{_mandir}/man1/curl-config.1*
+%{_mandir}/man3/*
+%{_datadir}/aclocal/libcurl.m4
+
+%changelog
+* Wed May 29 2024 Gael Duperrey <gduperrey@vates.tech> - 8.6.0-2.1
+- Synced from XS82ECU1063
+- Removed xenserver-specific test of the dist macro
```
Comment (stormi) [inline SPECS/curl.spec:347]:
```suggestion
- Removed xenserver-specific test of the dist macro
- Replaced perl-interpreter BuildRequires with perl
- Replaced unsupported %%lconfig_scriptlets macro with %%post and %%postun one liners
```

**Comment by gduperrey** [comment]
Koji build scratch to v8.2-incoming: https://koji.xcp-ng.org/taskinfo?taskID=54248

**Review by stormi — APPROVED** [review]

**Review by tescande — APPROVED** [review]

**Comment by gduperrey** [comment]
Koji build (v8.2-incoming): https://koji.xcp-ng.org/taskinfo?taskID=54319
